api/sse/decorators.py

```diff
@@ -0,0 +1,44 @@
+import typing
+
+from .tasks import send_identity_update_message
+
+
+def generate_identity_update_message(
+    get_data_from_req_callable=lambda req: (
+        req.environment,
+        req.data["identity"].get("identifier"),
+    )
+):
+    """
+    Warps a view method that send an identity update message if the view returns a 2XX response
+
+    Args:
+        get_data_from_req_callable (Callable): A callable that takes a request object and returns a tuple of
+            (environment, identifier)
+
+    Usage:
```

Comment:
I don't think this is quite right? It's using `method_decorator` directly on the method / function itself. It should just be: 

```python
@generate_identity_update_message(
    lambda req: (req.environment, req.data["identifier"])
)
```

right?

Reply:
No actually what we have is function decorator and in order to apply it on methods like `post`, `put` etc it needs to be converted to method decorator which is the job of `method_decorator`

Reply:
I don't think that's true. `method_decorator` exists so you can add the decorator to the method at the class level, instead of having to override the method and decorate it, you can still add the decorator directly to the method on the class if you want to? 

I think this simple script illustrates this:

```python
from django.utils.decorators import method_decorator


def my_decorator(foo: str):
    def outer(func):
        def inner(*args, **kwargs):
            print(f"{foo} before")
            func(*args, **kwargs)
            print(f"{foo} after!")
        return inner
    return outer


@my_decorator("decorated function")
def my_decorated_function():
    pass


@method_decorator(name="decorated_using_method_decorator", decorator=my_decorator("using method_decorator"))
class MyDecoratedClass:
    @my_decorator("decorated method directly")
    def decorated_directly(self):
        pass

    def decorated_using_method_decorator(self):
        pass


if __name__ == "__main__":
    my_decorated_function()

    o = MyDecoratedClass()
    o.decorated_directly()
    o.decorated_using_method_decorator()
```

Reply:
Note that the above prints out: 

```
decorated function before
decorated function after!
decorated method directly before
decorated method directly after!
using method_decorator before
using method_decorator after!
```

Reply:
Good point, but that will result in first argument being self instead of request, no? 

```
def method_decorator(decorator, name=""):
    """
    Convert a function decorator into a method decorator
    """
```
ref: https://github.com/django/django/blob/main/django/utils/decorators.py#L56